frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/contextmenu/SyncSelectedToParentByMergeAction.java

```diff
@@ -1,16 +1,50 @@
 package com.virtuslab.gitmachete.frontend.actions.contextmenu;
 
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import com.intellij.ide.util.PropertiesComponent;
 import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.ui.MessageDialogBuilder;
+import com.intellij.openapi.ui.Messages;
 import io.vavr.control.Option;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
 
 import com.virtuslab.gitmachete.frontend.actions.base.BaseSyncToParentByMergeAction;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.WarnAboutRebaseToParentByMergeDialog;
 import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeySelectedBranchName;
 
 public class SyncSelectedToParentByMergeAction extends BaseSyncToParentByMergeAction
     implements
       IExpectsKeySelectedBranchName {
+
+  public static final String MERGE_INFO_SHOWN = "git-machete.reset.info.shown";
```

Comment:
Hmmm isn't this key already used for the Reset warning (distinct from this one)? 🤔